pkg/crc/machine/hyperv/driver_windows.go

```diff
@@ -23,5 +29,32 @@ func CreateHost(machineConfig config.MachineConfig) *hyperv.Driver {
 		hypervDriver.VirtualSwitch = switchName
 	}
 
+	for _, dir := range machineConfig.SharedDirs {
+		sharedDir := drivers.SharedDir{
+			Source:   dir,
+			Target:   convertToUnixPath(dir),
+			Tag:      smbShareUNCPathFromTag("crc-dir0"), // smb share 'crc-dir0' is created in the msi
+			Type:     "cifs",
+			Username: machineConfig.SharedDirUsername,
+		}
+		hypervDriver.SharedDirs = append(hypervDriver.SharedDirs, sharedDir)
+	}
 	return hypervDriver
 }
+
+func convertToUnixPath(path string) string {
+	// converts a path like c:\users\crc to /mnt/c/users/crc
+	path = filepath.ToSlash(path)
+	if len(path) > 1 && path[1] == ':' {
+		return ("/mnt/" + strings.ToLower(path[0:1]) + path[2:])
+	}
+	return path
+}
+
+func smbShareUNCPathFromTag(tag string) string {
+	hostname, _, err := powershell.Execute("hostname")
+	if err == nil {
+		return fmt.Sprintf("//%s.mshome.net/%s", strings.TrimSpace(hostname), tag)
```

Comment:
This is an assumption that will only work for the default switch if not mistaken.
better to use the IP address or actual hostname for this. This can be extracted but is tedious

Reply:
well, the IP address also we have to use from the `Default Switch` no? or you mean enumerate all the interfaces and pick an ip address from there?

Reply:
btw, this hostname.mshome.net also works with the user-mode networking, i  think in that case the VM doesn't any ip assigned or connection with the default switch

Reply:
We do not bind the default switch to the VM, as we use the usermode networking.
We can not actually assume the default switch is there and enabled. Our preflight is erroneous, as we do not need it anymore... and does not check if enabled or even reachable.

Reply:
Isn't the IP not JUST the external IP of our usermode p2p ?

Reply:
`mshome.net` does not work for me. I do not have ICS enabled and DNS priorities wouldn't have allowed it anyway.

Reply:
okay, in that case we need to switch to using IP address, thanks for testing out.

> Isn't the IP not JUST the external IP of our usermode p2p ?

no able to understand this, what is that external IP of usermode networking? when i have the daemon running, it doesn't create any interfaces

Reply:
In usermode networking, I'd expect 127.0.0.1?

Reply:
`gvproxy` also has a "host-access" feature, where in the VM can reach the host using a dns name, by default we disabled it (can be enabled from config), I think that is exactly what we need

```
* host-network-access                     Allow TCP/IP connections from the CRC VM to services running on the host (true/false, default: false)
```

Reply:
host access might be able to expose port `445` IIRC, but that would make this available on the VM's localhost interface to be the SMB endpoint. Don't think this is what you want. Somehow you want to forward the SMB functionality to be resolvable.